homeassistant/components/switch/mochad.py

```diff
@@ -59,17 +64,35 @@ def name(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on."""
-        self._state = True
         with mochad.REQ_LOCK:
-            self.device.send_cmd('on')
-            self._controller.read_data()
+            try:
+                self._state = True
+                # Recycle socket on new command to recover mochad connection
+                _LOGGER.debug("Reconnect %s:%s", self._controller.server,
+                                self._controller.port)
+                self._controller.reconnect()
+                self.device.send_cmd('on')
+                # No read data on CM19A which is rf only
+                if (self._comm_type == 'pl'):
+                    self._controller.read_data()
+            except Exception as e:
```

Comment:
Don't except all exceptions. Import `MochadException` from `pymochad.exceptions` and except that exception and `OSError` to catch socket errors.

Also change variable name to eg `exc`.

Reply:
Ok. Fixed in commit 9fbade8e7